plutus-playground/plutus-playground-server/src/Playground/Interpreter.hs

```diff
@@ -1,166 +1,195 @@
-import           System.Directory             (removeFile)
-import           System.IO.Temp               (writeSystemTempFile)
-import           Wallet.Emulator.Types        (EmulatorEvent, Wallet)
-
-$(TH.mkFunction 'payToPublicKey_)
-
-defaultExtensions :: [Extension]
-defaultExtensions =
-    [ DataKinds
-    , DeriveAnyClass
-    , DeriveFoldable
-    , DeriveFunctor
-    , DeriveGeneric
-    , DeriveLift
-    , DeriveTraversable
-    , ExplicitForAll
-    , FlexibleContexts
-    , OverloadedStrings
-    , RecordWildCards
-    , ScopedTypeVariables
-    , StandaloneDeriving
-    , TemplateHaskell
-    ]
-
-loadSource :: (MonadInterpreter m) => FilePath -> (ModuleName -> m a) -> m a
-loadSource fileName action =
-    flip finally (liftIO (removeFile fileName)) $ do
-        set [languageExtensions := defaultExtensions]
-        loadModules [fileName]
-        (m:_) <- getLoadedModules
-        setTopLevelModules [m]
-        action m
+import           Control.Monad              (unless)
+import           Control.Monad.Catch        (MonadCatch, MonadMask, bracket, catch)
+import           Control.Monad.Error.Class  (MonadError, throwError)
+import           Control.Monad.IO.Class     (MonadIO, liftIO)
+import qualified Control.Newtype.Generics   as Newtype
+import           Data.Aeson                 (ToJSON)
+import qualified Data.Aeson                 as JSON
+import           Data.ByteString            (ByteString)
+import qualified Data.ByteString.Char8      as BS8
+import qualified Data.ByteString.Lazy.Char8 as BSL
+import           Data.List                  (intercalate)
+import           Data.Swagger               (Schema)
+import           Data.Text                  (Text)
+import qualified Data.Text                  as Text
+import qualified Data.Text.Internal.Search  as Text
+import qualified Data.Text.IO               as Text
+import           Ledger.Types               (Blockchain, Value)
+import           Playground.API             (Evaluation (sourceCode), Expression (Action, Wait), Fn (Fn),
+                                             FunctionSchema,
+                                             PlaygroundError (CompilationErrors, DecodeJsonTypeError, InterpreterError, OtherError),
+                                             SourceCode, parseErrorsText, program, wallets)
+import           System.Directory           (removeFile)
+import           System.Environment         (lookupEnv)
+import           System.Exit                (ExitCode (ExitSuccess))
+import           System.IO                  (Handle, hClose, hFlush)
+import           System.IO.Temp             (getCanonicalTemporaryDirectory, openTempFile)
+import           System.Process             (readProcessWithExitCode)
+import qualified Text.Regex                 as Regex
+import           Wallet.Emulator.Types      (EmulatorEvent, Wallet)
+
+run :: (MonadIO m, MonadError PlaygroundError m) => FilePath -> m String
```

Comment:
unused?

Reply:
Seems like this and `runscript` should be merged.

Reply:
already removed, about to push